Worksheet date_modified tests are flaky

The worksheet date_modified tests are flaky because they check for an interval of 1s. This makes test failures such as in this run (https://github.com/codalab/codalab-worksheets/runs/1670895176):

```
FAIL: test_create (tests.unit.rest.worksheets_test.WorksheetsTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/codalab-worksheets/tests/unit/rest/worksheets_test.py", line 38, in test_create
    self.assertAlmostEqual(current_time, date_created, delta=datetime.timedelta(seconds=1))
AssertionError: datetime.datetime(2021, 1, 8, 20, 19, 29, 656770) != datetime.datetime(2021, 1, 8, 20, 19, 28) within datetime.timedelta(0, 1) delta
```

--- a/tests/unit/rest/worksheets_test.py
+++ b/tests/unit/rest/worksheets_test.py
@@ -26,8 +26,8 @@ class WorksheetsTest(BaseTestCase):
                     "relationships": {
                         "items": {
                             "data": [
-                                {"type": "worksheet-items", "id": f"('name', '{worksheet_name}')",},
-                                {"type": "worksheet-items", "id": f"('uuid', '{worksheet_id}')",},
+                                {"type": "worksheet-items", "id": f"('name', '{worksheet_name}')"},
+                                {"type": "worksheet-items", "id": f"('uuid', '{worksheet_id}')"},
                             ]
                         }
                     },

--- a/tests/unit/rest/worksheets_test.py
+++ b/tests/unit/rest/worksheets_test.py
@@ -1,14 +1,18 @@
 import uuid
 from .base import BaseTestCase
 import datetime
+from freezegun import freeze_time
 
 
 class WorksheetsTest(BaseTestCase):
-    def test_create(self):
+    @freeze_time("2012-01-14", as_kwarg="frozen_time")
+    def test_create(self, frozen_time):
+        """Create a new worksheet, then ensure that the proper fields are returned.
+        """
         worksheet_name = f"codalab-{uuid.uuid4()}"
         response = self.app.post_json(
-            '/rest/worksheets',
-            {'data': [{'type': 'worksheets', 'attributes': {'name': worksheet_name}}]},
+            "/rest/worksheets",
+            {"data": [{"type": "worksheets", "attributes": {"name": worksheet_name}}]},
         )
         worksheet_id = response.json["data"][0]["id"]
         data = response.json["data"]
@@ -22,18 +26,17 @@ class WorksheetsTest(BaseTestCase):
                     "relationships": {
                         "items": {
                             "data": [
-                                {"type": "worksheet-items", "id": f"('name', '{worksheet_name}')"},
-                                {"type": "worksheet-items", "id": f"('uuid', '{worksheet_id}')"},
+                                {"type": "worksheet-items", "id": f"('name', '{worksheet_name}')",},
+                                {"type": "worksheet-items", "id": f"('uuid', '{worksheet_id}')",},
                             ]
                         }
                     },
                 }
             ],
         )
-        # Verify the date_created and date_last_modified field has been set to the current time for a new created worksheet
-        worksheet_info = self.app.get('/rest/interpret/worksheet/' + worksheet_id).json
-        date_created = datetime.datetime.strptime(worksheet_info['date_created'], '%Y-%m-%dT%H:%M:%S')
-        date_last_modified = datetime.datetime.strptime(worksheet_info['date_last_modified'], '%Y-%m-%dT%H:%M:%S')
-        current_time = datetime.datetime.utcnow()
-        self.assertAlmostEqual(current_time, date_created, delta=datetime.timedelta(seconds=1))
-        self.assertAlmostEqual(current_time, date_last_modified, delta=datetime.timedelta(seconds=1))
+
+        # Verify that the date_created and date_last_modified fields are set to the current time for a newly created worksheet.
+        worksheet_info = self.app.get("/rest/interpret/worksheet/" + worksheet_id).json
+        current_time = datetime.datetime.isoformat(frozen_time.time_to_freeze)
+        self.assertEqual(current_time, worksheet_info["date_created"])
+        self.assertEqual(current_time, worksheet_info["date_last_modified"])

